### ISSUE TITLE
Android 'debug' variant not found in KMM + Android project

### ISSUE THREAD

**Issue by alvr** [Kotlin/kotlinx-kover#388]
**Describe the bug**
When applying the plugin to the root project and specified subprojects, and using `mergeWith` in the default configuration, it throws and exception that the 'debug' Android variant was not found. Kover executes the tests that are in `jvmTest` source set (no `commonTest` yet), I suppose that when using `mergeWith` with the Android variant will execute also the tests in `androidTest` and `commonTest`.

**Errors**
`Android build variant 'debug' was not found - it is impossible to add it to the default report`

**Expected behavior**
Include the debug variant of Android in a KMM project.

**Reproducer**
The configuration of Kover is [here](https://github.com/alvr/katana/blob/feature/kmm/stable/build-logic/gradle-convention/src/main/kotlin/dev/alvr/katana/buildlogic/gradle/KatanaKoverPlugin.kt), and the only variants the project has are 'debug', 'beta' and 'release'. The full project is in [feature/kmm/stable](https://github.com/alvr/katana/tree/feature/kmm/stable).

**Environment**
 - Kover Gradle Plugin version: 0.7.0
 - Gradle version: 8.1.1
 - Kotlin project type: Kotlin/Multiplatform + Android
 - Coverage Toolset (if customized in build script): Kover

**Comment by sabufung30** [comment]
It looks like a duplicate of https://github.com/Kotlin/kotlinx-kover/issues/362
Is there any workaround by the moment ? @shanshin

**Comment by alvr** [comment]
I saw that issue and tried to put all the `Project.configure()` inside `afterEvalute` and I get the same error.

**Comment by shanshin** [comment]
> It looks like a duplicate of #362

Yes, most likely a duplicate

> Is there any workaround by the moment ?

@alvr, @sabufung30, сould you change the configuration so that the kover plugin is applied only after the android plugin?

In this case, the Android plugin will create build variants before the Kover plugin starts searching for them.

**Comment by sabufung30** [comment]
@shanshin @alvr 
I have put the configuration from `subprojects` into every single `build.gradle.kts` and it works.

**Comment by shanshin** [comment]
@sabufung30 the order of applying of plugins is important. By moving the configuration from `subprojects`, you most likely changed the order and the Kover plugin started being applied after Android Gradle Plugin.
This is one of the solutions.

**Comment by shanshin** [comment]
Closing as a duplicate.
Follow the progress in correcting the error in the #362 task.
